apps/aecore/lib/aecore/keys/keys.ex

```diff
@@ -0,0 +1,207 @@
+defmodule Aecore.Keys do
+  @moduledoc """
+  Module for handling the Wallet (signing) keys and Peer keys.
+  Keys are created on first run of the project and saved in respective dirs.
+  """
+
+  alias Aeutil.Bits
+  alias Aecore.Chain.Identifier
+
+  @typedoc "Defines what type of keypair we could have"
+  @type keypair_type :: :sign | :peer
+
+  @typedoc "Public key for signing or for peers - 32 bytes in size"
+  @type pubkey :: binary()
+
+  @typedoc "Private key for signing - 64 bytes in size"
+  @type sign_priv_key :: binary()
+
+  @typedoc "Private key for peers - 32 bytes in size"
+  @type peer_priv_key :: binary()
+
+  @type sign_keypair :: {pubkey(), sign_priv_key()}
+  @type peer_keypair :: {pubkey(), peer_priv_key()}
+  @type message :: binary()
+  @type signature :: binary()
+
+  @pub_size 32
+
+  @peer_key_encode "pp"
+  @peer_key_decode "pp$"
+
+  @filename_sign_pub "sign_key.pub"
+  @filename_sign_priv "sign_key"
+
+  @filename_peer_pub "peer_key.pub"
+  @filename_peer_priv "peer_key"
+
+  @doc """
+  Returns a signed version of the given binary
+  """
+  @spec sign(message()) :: signature()
+  def sign(message) when is_binary(message) do
+    {_, privkey} = keypair(:sign)
+    sign(message, privkey)
+  end
+
+  @spec sign(message(), sign_priv_key()) :: signature()
+  def sign(message, privkey) when is_binary(message) and is_binary(privkey) do
+    :enacl.sign_detached(message, privkey)
+  end
+
+  @doc """
+  Checks if the message is signed with the given public key
+  """
+  @spec verify(message(), signature()) :: true | false
+  def verify(message, sign) do
+    {pubkey, _} = keypair(:sign)
+    verify(message, sign, pubkey)
+  end
+
+  @spec verify(message(), signature(), pubkey()) :: true | false
+  def verify(message, sign, pubkey)
+      when is_binary(message) and is_binary(sign) and is_binary(pubkey) do
+    case :enacl.sign_verify_detached(sign, message, pubkey) do
+      {:ok, _} -> true
+      _ -> false
+    end
+  end
+
+  @spec key_size_valid?(binary()) :: true | false
+  def key_size_valid?(%Identifier{value: pubkey})
+      when byte_size(pubkey) == @pub_size,
+      do: true
+
+  def key_size_valid?(pubkey)
+      when byte_size(pubkey) == @pub_size,
+      do: true
+
+  def key_size_valid?(_), do: false
+
+  @spec keypair(keypair_type(), String.t()) :: sign_keypair() | peer_keypair()
+  def keypair(type, suffix \\ "") do
```

Comment:
Can you split this into Keys.sign_pubkey, Keys.peer_pubkey and so on for testing purposes

Reply:
This was made intentionally, to keep the code as small and simple as possible. For testing just add an atom `:sign | :peer` to get the specific keypair

Reply:
It's better to have Keys.sign_pubkey instead of typing out Keys.keypair(:sign) |> elem(0) every time, people might not know at what position the key they want is

Reply:
Then I'll add to the `doc` of the function. I don't feel like changing a function for the sake of writing test easier.